### ISSUE TITLE
[BUG] Forecasting horizon in score method

### ISSUE THREAD

**Issue by abdkg** [alan-turing-institute/sktime#167]
**Describe the bug**
The following code fails when no forecasting horizon is provided to compute score.

**To Reproduce**
```python
from sktime.forecasters import ExpSmoothingForecaster
from sktime.datasets import load_shampoo_sales

import pandas as pd


shampoo = load_shampoo_sales()

# split data into train, update and test
train = pd.Series([shampoo.iloc[0].iloc[:30]])
test = pd.Series([shampoo.iloc[0].iloc[30:]])

# simple exponential smoothing
m = ExpSmoothingForecaster()
m.fit(train)
m.score(test)
```

**Expected behavior**
Code should return a score

**Additional context**
Remove `None` as default fh

**Versions**
<details>

Darwin-18.6.0-x86_64-i386-64bit
Python 3.7.4 (default, Aug  5 2019, 21:56:28) 
[Clang 10.0.1 (clang-1001.0.46.4)]
NumPy 1.17.2
SciPy 1.3.1
sktime 0.3.0

</details>

**Comment by mloning** [comment]
Thanks for reporting the bug! I believe this has been fixed already (see #126) but isn't in the latest release yet.

**Comment by abdkg** [comment]
Hello @mloning,

I've already seen this PR. Tell me if I'm wrong, but it seems to me that the issue was fixed for `fix`and `predict` methods, but not `score`.
Also I left out than I ran my tests on the dev branch.

**Comment by mloning** [comment]
Yes, you're right, thanks, will fix this!

**Comment by mloning** [comment]
Yes, you're right, thanks! Will add this via the forecasting branch, unless you want to make a PR?

**Comment by abdkg** [comment]
Thank you for proposing. I made a new PR.

**Comment by mloning** [comment]
Fixed by #168
